### PR TITLE
psyad unary minus fix [closes 1583]

### PR DESCRIPTION
**Pull request by rupertford**

### CODE REVIEW

File: src/psyclone/tests/psyad/transformations/test_assignment_trans.py [hunk 1]
```diff
@@ -1157,6 +1175,30 @@ def test_validate_rhs_active_multi_divisor():
     trans.validate(assignment)
 
 
+def test_validate_rhs_active_unary_minus():
+    '''Test that the validation works when there is a unary minus on
+    the lhs of an expression.
```
Comment (arporter) [inline src/psyclone/tests/psyad/transformations/test_assignment_trans.py:1180]: s/an expression/a rhs expression/

File: src/psyclone/tests/psyad/transformations/test_assignment_trans.py [hunk 2]
```diff
@@ -526,6 +526,24 @@ def test_multi_increment(tmpdir, index_str):
     check_adjoint(tl_fortran, active_variables, ad_fortran, tmpdir)
 
 
+def test_unary_minus(tmpdir):
+    '''Test that the transformation works when there is a unary minus on
+    the lhs of an expression.
```
Comment (arporter) [inline src/psyclone/tests/psyad/transformations/test_assignment_trans.py:531]: s/an expression/a rhs expression/

**Review by arporter — CHANGES_REQUESTED** [review]
Fix is good.
All tests and examples are fine with compilation.
I just found two of the comments a bit confusing so please could you reword slightly.
Optionally, would you mind fixing the f-string warnings in `psyad/transformations/assignment_trans.py`?

File: src/psyclone/tests/psyad/transformations/test_assignment_trans.py (hunk 1 above)
Comment (rupertford) [reply src/psyclone/tests/psyad/transformations/test_assignment_trans.py:1183]: modified

File: src/psyclone/tests/psyad/transformations/test_assignment_trans.py (hunk 2 above)
Comment (rupertford) [reply src/psyclone/tests/psyad/transformations/test_assignment_trans.py:532]: modified.

**Comment by rupertford** [comment]
I've made the requested changes and brought the branch up to master. There are no warnings in `psyad/transformations/assignment_trans.py`. I'm guessing an earlier PR has already sorted these out. Ready for next review from @arporter.

**Review by arporter — CHANGES_REQUESTED** [review]
All requested changes have been made.
Tests and examples still all OK with compilation.
This is a bug fix so no documentation update required.
Will proceed to merge.

**Review by arporter — APPROVED** [review]

**Pull request merged**